astro/src/content/quickstarts/quickstart-javascript-express-web.mdx

```diff
@@ -0,0 +1,320 @@
+page. To see the client ID and client secret, go to the Applications page and 
+click the `View` icon under the actions for the Changebank application. You'll 
+find the client id and client secret values in the `OAuth configuration` section.
+</Aside>
+
+#### Static files
+
+We need a way to serve our static files, luckily Express makes this easy with `express.static`. These files will include all of the CSS and images that are required for the UI of the ChangeBank application.
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="static"/>
+
+#### Homepage
+
+In order to show off our homepage we need to create a route with the base url `/`. In this route check to see if `userDetails` cookie has any user values and correctly respond with either the `home.html` or `account.html` template. Also if there is no user you can generate the PKCE challenge pair and store them into `userSession` cookie (take note this is an http-only cookie).
+
+<Aside type="note">
+In this file we will use the package [pkce-challenge](https://www.npmjs.com/package/pkce-challenge) to generate a Proof Key for Code Exchange (PKCE) challenge pair. Read more about [PKCE](https://www.oauth.com/oauth2-servers/pkce/authorization-request/).
+</Aside>
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="homepage"/>
+
+#### Login Route
+
+In the Changebank application there is a `Login` button, when clicked will open the `/login` route (you can find this in `complete-application/templates/home.html`). Have this redirect them to FusionAuth so that FusionAuth can present them with a login page.
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="login"/>
+
+You can always find this url if you view the details of your Application within FusionAuth, under "OAuth2 & OpenID Connect Integration details" you will find the property `OAuth IdP login URL`. The endpoint that you will hit on FusionAuth is `/oauth2/authorize` below you can find a list of the needed parameters for FusionAuth's [OAuth Endpoints](https://fusionauth.io/docs/v1/tech/oauth/endpoints)
+
+- client_id: Your `applicationId`
+- response_type: `code` since we are using the code flow
+- redirect_uri: Where your FusionAuth will redirect back to during the Oauth flow, in our example this is our route `oauth-redirect`
+- state: Created in our homepage route using some crazy math and stored in our cookie as `stateValue`
+- code_challenge: Created in our homepage route using `pkce-challenge` stored in our cookie as `challenge`, this is one half of the PKCE 
+
+#### OAuth Redirect
+
+When FusionAuth redirects back to our Express application it will return to our `/oauth-redirect` route. In this response it will have two query parameters that we need to further examine `code` and `state`. 
```

Comment:
```suggestion
When FusionAuth redirects back to our Express application, it will return to our `/oauth-redirect` route. In this response it will have two query parameters that we need to further examine: `code` and `state`. 
```

Not sure about commas again